src/sentry/utils/snuba.py

```diff
@@ -448,3 +450,17 @@ def insert_raw(data):
             )
     except urllib3.exceptions.HTTPError as err:
         raise SnubaError(err)
+
+
+def shrink_time_window(issues, start, end):
+    if issues and len(issues) == 1:
+        group_id = issues[0][0]
+        group = Group.objects.get(pk=group_id)
+        start = max(start, naiveify_datetime(group.first_seen) - timedelta(minutes=5))
```

Comment:
Do we need to guard against first_seen or last_seen being null or are they always guaranteed to be defined

Reply:
Yeah they're not nullable (and set on `save`).